app/src/features/apiClient/screens/apiClient/utils.ts

```diff
@@ -184,3 +199,113 @@ export const convertFlatRecordsToNestedRecords = (records: RQAPI.Record[]) => {
   return updatedRecords;
 };
 export const getEmptyPair = (): KeyValuePair => ({ id: Math.random(), key: "", value: "", isEnabled: true });
+
+export const getEmptyAuthOptions = (): AUTH_OPTIONS => {
+  return {
+    [AUTHORIZATION_TYPES.NO_AUTH]: null,
+    [AUTHORIZATION_TYPES.API_KEY]: { key: "", value: "", addTo: "HEADER" },
+    [AUTHORIZATION_TYPES.BEARER_TOKEN]: { bearer: "" },
+    [AUTHORIZATION_TYPES.BASIC_AUTH]: { username: "", password: "" },
+  };
```

Comment:
This data looks unnecessary while saving to db. All the extra form data except the selected one should get stripped away. 

We can keep the previous state in the AuthorizationForm component only to keep the last know values while switching between different Auths. But the final onChange should only send the value for the selected auth type only